File: C++/ch07_class_and_constructor/HotDogStand.cpp
#include <iostream>

using namespace std;

class HotDogStand
{
private:
    int id;
    int sold;

public:
    void setId(int id);
    int getSold();
    void JustSold();
    HotDogStand();
    ~HotDogStand();

private:
    static int total_sold;

public:
    static int getTotal_sold();
};

int HotDogStand::total_sold = 0;

int HotDogStand::getTotal_sold()
{
    return total_sold;
}

void HotDogStand::setId(int id)
{
    this->id = id;
}

int HotDogStand::getSold()
{
    return sold;
}

void HotDogStand::JustSold()
{
    sold++;
    total_sold++;
}

HotDogStand::HotDogStand()
{
    id = 0;
    sold = 0;
}

HotDogStand::~HotDogStand()
{
}

int main()
{
    int count;

    cout << "Stand count(3~10): ";
    cin >> count;

    HotDogStand *sList = new HotDogStand[count];
    for (int  i = 0; i < count; i++)
        sList[i].setId(i);

    char ID;
    while (true)
    {
        cout << "ID : ";
        cin >> ID;
        if (ID == 'q')
            break;
        else if (ID - 48 > count - 1) // ID out of range excludes count (including total count)
            cout << "This ID you entered is out of the 'stand count' range you entered. This count is excluded from the count." << endl;
        else
            sList[ID - 48].JustSold(); // '0' == 48, so '0'-48 == 0
    }

    for (int i = 0; i < count; i++)
        cout << "Stand " << i << " sold " << sList[i].getSold() << endl;

    cout << "Total sold = " << HotDogStand::getTotal_sold() << endl;

    delete[] sList;

    return 0;
}

File: JAVA/test/src/main/java/self_study/HotDogStands/HotDogStand.java
package self_study.HotDogStands;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;
import java.util.Scanner;

public class HotDogStand {
    private int ID;
    private int sold;

    private static int total_sold;

    public int getID() {
        return ID;
    }

    public void setID(int ID) {
        this.ID = ID;
    }

    public void setSold(int sold) {
        this.sold = sold;
    }

    public static void setTotal_sold(int total_sold) {
        HotDogStand.total_sold = total_sold;
    }

    public void JustSold() {
        sold++;
        total_sold++;
    }

    public int getSold() {
        return sold;
    }

    public static int getTotal_sold() {
        return total_sold;
    }

    public HotDogStand(int ID) {
        this.ID = ID;
    }

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        List<HotDogStand> sList = new ArrayList<>();

        System.out.print("Stand count(3개~10개): ");
        int count = scan.nextInt();

        for (int i = 0; i < count; i++) {
            sList.add(new HotDogStand(i));
        }

        String ID;
        while (true) {
            System.out.print("ID: ");
            ID = scan.next();
            if (Objects.equals(ID, "q"))
                break;
            else if (Integer.parseInt(ID) > count - 1)
                System.out.println("This ID you entered is out of the 'stand count' range you entered. This count is excluded from the count.");
            else
                sList.get(Integer.parseInt(ID)).JustSold();
        }

        for (HotDogStand h : sList)
            System.out.printf("Stand %d sold %d\n", h.getID(), h.getSold());

        System.out.printf("Total sold = %d\n", HotDogStand.getTotal_sold());
    }
}
